Базовая стилизация всех страниц

alina-kh opened the pull request:
---
:mortar_board: [Базовая стилизация всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2299541/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/2299541-gllacy-36/1/

alina-kh committed:
Добавляет наставника

alina-kh committed:
Добавляет разметку страниц

alina-kh committed:
Добавляет изображения для сайта

alina-kh committed:
Добавляет разметку - поправила верстку

alina-kh committed:
Добавляет картинки

alina-kh committed:
Добавляет стили

alina-kh commented:
Добавляет
наставника
разметку
картинки
общие стили

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+            </div>
+            <img class="slider__image" src="images/slider/ice-1.png" alt="Нежный пломбир с клубничным джемом" width="328" height="507">
+            <img class="slider__image--next" src="images/slider/ice-2-small.png" alt="Сливочное мороженое со вкусом банана" width="66" height="150">
+            <img class="slider__image--next" src="images/slider/ice-3-small.png" alt="Карамельный пломбир с маршмеллоу" width="66" height="150">
```
![image](https://user-images.githubusercontent.com/20788716/214922900-67191d16-5d55-41cd-bda8-86c76ee0d043.png)
Модификатор может находится только рядом с классом, от которого образован.

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+            </div>
+            <img class="slider__image" src="images/slider/ice-2.png" alt="Сливочное мороженое со вкусом банана" width="328" height="507">
+            <img class="slider__image--next" src="images/slider/ice-3-small.png" alt="Карамельный пломбир с маршмеллоу" width="66" height="150">
+            <img class="slider__image--next" src="images/slider/ice-1-small.png" alt="Нежный пломбир с клубничным джемом" width="66" height="150">
```
Не нужно в каждом слайдер размечать по три картинки. 
Тут другая система, в каждом слайдер текст и одна картинка, когда слайдер не активных, то текст скрывается,  а картинка становится маленькой.
Можно разделить на два связанных слайдера, один будет с текстом, где слайды пропадают, а второй с картинка, где активная большая и впереди всех.

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+          <li class="dot dot--current"></li>
+          <li class="dot"></li>
+          <li class="dot"></li>
+        </ul>
```
Если исходить из БЭМ, я бы советовала сделать блок slider, в который входит обертка слайдера, сами слайды (slider__item) и навигация слайдера, т.е. стрелочки и кнопки.
Все, что внутри слайдера к нему не относится - это содержимое карточек, оно может меняться,  к примеру, если бы у тебя на сайте было 5 слайдеров, они бы все были с разным содержимым, но имели бы одинаковые кнопочки и способы переключения, поэтому slider - это переиспользуемый блок в разных сладерах, а содержимое - это отдельные блоки.

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+          <li class="dot"></li>
+          <li class="dot"></li>
+        </ul>
+        <ul class="social slider-social--footer">
```
Модификатор --footer не очень подходит, учитывая где находится блок.
К тому же класс slider-social--footer, может стоят только рядом с классом slider-social

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+          <li class="gifts__item gifts__item--one">
+            <h4 class="gifts__subtitle title-h4">Малинка даром!</h4>
+              <p class="gifts__text text">При покупке 2 кг любого фруктового мороженого добавим в ваш заказ банку малинового варенья бесплатно.</p>
+              <a class="button button--white">Хочу подарок</a>
```
У ссылок должен быть атрибут href

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+      <section class="popular">
+        <h3 class="popular__title title-h3">Попробуйте самые популярные вкусы нашего мороженого</h3>
+        <ul class="popular__list">
+          <li class="popular__item">
```
В каталоге находятся точно такие же карточки, и там они под классом catalog. Т.е. это те же самые блоки, которые переиспользуются на главной странице, не нужно их размечать под другим классом.

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+              <li class="info__item info__item--three">
+                Для приготовления мороженого используются сливки и молоко высочайшего качества. Все ингредиенты и добавки произведены из натурального сырья.
+              </li>
+              <li class="info__item info__item--four">
```
Не очень хорошие модификаторы, не объясняют для чего они нужны, разве что эти модификаторы добавляют цифры, но это не так, поэтому лучше называть по картинкам или даже прописать, что это именно для добавления иконок.

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+      </section>
+      <section class="news">
+        <p class="news__text">Новое в нашем блоге</p>
+        <h4 class="news__title title-h4">10 способов сервировки фруктовых щербетов к столу</h4>
```
Ссылка на блог.

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+      </section>
+    </main>
+    <footer class="page-footer">
+      <ul class="social slider-social--footer">
```
Модификатор может использоваться только в паре с классом, от которого образован

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+      <a class="page-footer__logo" href="https://htmlacademy.ru/" target="_blank">
+        <img src="images/icons/logo_htmlacademy.svg" alt="Логотип html academy" width="93" height="27">
+      </a>
+      <span class="page-footer__descr">Сделано в .html academy © 2022</span>
```
Надпись ".html academy" тоже ссылка, она выделена подчеркиванием

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,321 @@
+            <a class="breadcrumds__link" href="#">Каталог</a>
+          </li>
+          <li class="breadcrumds__item">
+            <a class="breadcrumds__link breadcrumds__link--selected">Сливочное мороженое</a>
```
Не ошибка, но модификатор на хлебных крошках обычно не ставиться, потому что пункт текущей страницы всегда последний, удобно стилизовать с помощью ` :last-child`

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,321 @@
+            <legend class="filter-form__title text--small">Жирность:</legend>
+            <ul class="filter-form__list">
+              <li class="filter-form__item">
+                <label class="control text--small">
```
Модификатор может использоваться только в паре с классом, от которого он образован
![image](https://user-images.githubusercontent.com/20788716/214928054-9e6a2cf0-8233-4a60-890b-d7c784141e2a.png)

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,271 @@
+          <li class="popular__item">
+            <a class="popular__link" href="">
+              <img class="popular__image" src="images/catalog/ice-1.png" alt="Малинка" width="168" height="168">
+              <h5 class="popular__subtitle title-h5">Малинка</h5>
```
Давай разберем заголовки.
У страницы заголовок h1 (скрытый, видимый - не важно, главное, чтобы он был), у секшенов заголовки h2 (все секшены стоят рядом друг с другом, значит, у них заголовок на одни уровень ниже, чем заголовок страницы, у блоков/элементов, что внутри секшеном заголовки h3, т.е. идет на пониже на один уровень.
Откуда же взялся заголовок h5?
Можно раздавать классы по размеру заголовка, потому что по классам идет стилизация, то теги нужно ставить по порядку, т.е.
```
<div>
  <h1></h1>
  <div>
    <h2></h2>
    <div>
      <h3></h3>
    </div>
  </div>
  <div>
    <h2></h2>
    <div>
      <h3></h3>
      <div>
        <h4></h4>
      </div>
    </div>
    <div>
      <h3></h3>
    </div>
  </div>
</div>

```

Komnatnaya commented:
**Далее правки по графике.**

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214930245-fa383f98-3cb9-4542-8945-65288f17f1f0.png)
Лучше вырезать в jpg и скруглять стилями

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214930532-9579ba0f-f685-4024-9932-c41c1125111a.png)
Названия изображений должны быть оформлены однообразно, в некоторых картинка используется тире для разделения слов, а в некоторых подчеркивание, нужно чтобы было одинаково.

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214930877-a78d13f2-0415-4388-b0ef-dbea116ecc3c.png)
Лишние изображения, уменьшать и делать частично прозрачными будем стилями.

Komnatnaya commented:
**Далее правки по стилизации**

Komnatnaya commented on styles/styles.css:
```diff
@@ -0,0 +1,455 @@
+}
+
+ul {
+  list-style: none;
```
Глобальные стили запрещены критерием, допустимые можно посмотреть в критерии https://up.htmlacademy.ru/htmlcss/36/criteries#CSS-02

Komnatnaya commented on styles/styles.css:
```diff
@@ -0,0 +1,455 @@
+}
+
+.visually-hidden {
+  display: none;
```
Вся суть доступного скрытия в том, чтобы не использовать display: none;)
```
.visually-hidden {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  border: 0;
  padding: 0;
  white-space: nowrap;
  clip-path: inset(100%);
  clip: rect(0 0 0 0);
  overflow: hidden;
}
```

Komnatnaya commented on styles/styles.css:
```diff
@@ -0,0 +1,455 @@
+}
+
+
+input {
```
Глобальные стили, нужно по классу  или по классу родителя и тегу.

Komnatnaya commented on styles/styles.css:
```diff
@@ -0,0 +1,455 @@
+  height: 32px;
+  background: #FF2F64;
+  border-radius: 26px;
+  background-image: url("/images/icons/cart-white.svg");
```
Слеш (/) это выход в корень, но проект не всегда находится в корне сайта, тут зависит многое от сборки, и порой адреса начинающийся со слеша оптимальный, но не в нашем случае. 
Если ты посмотришь на опубликованный пулреквест, то заметишь, что там нет фоновых картинок, потому что корень сайта в публикации гораздо выше и там нет папки images. Поэтому давай использовать относительные адреса, а именно выход из папки стилей и вход в папку images "../images"

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214935035-6d493897-3f70-423f-92e0-59fb39359159.png)
Цвет текста в кнопках не по макету.

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214935320-90d5c8be-f1e8-4cf7-a981-d3f6233a75de.png)
Цвет и подчеркивание не по макету, у Фисташки наверно класс не тот, она сместилась

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214935577-4be22bec-e936-4737-bc05-b5641cd47a8e.png)
Цвет нежирного текста на белом фоне в макете. Дальше именно его не буду указывать, на белом фоне он такой везде.

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214936534-d60ca94c-65f1-47df-8e7d-aeea9b8e0a83.png)
Кажется в макете другой номер, подчеркивание можно убрать

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214936639-f7a755d0-0e08-4a8f-9667-7869c514672f.png)
Жирным должен быть только первый пункт, на это можно добавить модификатор

Komnatnaya commented:
![image](https://user-images.githubusercontent.com/20788716/214937185-9ae75c37-5194-43b3-99dc-f0a14ba2382b.png)
Цвет в кнопках

alina-kh committed:
Исправляет замечания по разметке

alina-kh committed:
Исправляет замечания по графике

alina-kh committed:
Исправляет замечания по контенту

Исправляет расширение картинок в каталоге, добавляет общий класс импутам и спискам

alina-kh committed:
Исправляет замечания по стилям.

alina-kh committed:
Добавляет скругление картинкам

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,266 @@
+      </section>
+      <section class="news">
+        <p class="news__text text">Новое в нашем блоге</p>
+        <h2 class="news__title title-h4"><a class="link" href="#"></a>10 способов сервировки фруктовых щербетов к столу</a></h2>
```
Валидатор ругается, два раза закрыта ссылка.

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,266 @@
+              <p class="card__text text text--big">Натуральное мороженое из свежих сливок и молока с вкуснейшим клубничным джемом – это идеальный десерт для всей семьи.</p>
+              <button class="button button--white" type="button">Заказать</button>
+            </div>
+            <img class="card__image" src="images/slider/ice-1.png" alt="Нежный пломбир с клубничным джемом" width="328" height="507">
```
Ошибки по БЭМ.
Элементы за пределами блока ![image](https://user-images.githubusercontent.com/20788716/215168140-a0989d30-9eca-4250-920f-39ac3ce06c4d.png)

Komnatnaya commented on index.html:
```diff
@@ -3,10 +3,266 @@
+        <ul class="slider__dots list">
+          <li class="dot dot--current"></li>
+          <li class="dot"></li>
+          <li class="dot"></li>
```
Это кнопки. Если элемент интерактивный, то это либо ссылки, либо кнопки.

Komnatnaya commented:
**С графикой все хорошо**

Komnatnaya commented:
> ![image](https://user-images.githubusercontent.com/20788716/214936534-d60ca94c-65f1-47df-8e7d-aeea9b8e0a83.png) Кажется в макете другой номер, подчеркивание можно убрать

Я не ту картинку положила, правка была про номер телефона 
![image](https://user-images.githubusercontent.com/20788716/215171092-d8ccc6d3-2907-4cba-9fa3-69d27cd70be9.png)

alina-kh committed:
Исправляет замечания по разметке
